### PR TITLE
Fixes some issues with traps

### LINKED ISSUE #1900: Blastmine and Claymore Trap behavior
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Issue Prelude

[//]: # (Please complete these mandatory steps and check the following boxes by putting an `x` inside the brackets _before_ filing your issue)

-  [x ] I have not modified the source prior to reproducing this issue.
-  [x ] I am using the latest version of Hercules.
-  [ x] I am aware that this report will be closed or deleted if it becomes obvious that I am stating the false.

### Description
Hercules: All monsters caught on the same trap receives the whole damage regardless of their number.

Official/Aegis: Damage taken by the monsters on a single trap is divided depending on the number of monsters that is caught.
This is a behavior of Pre-Renewal and Renewal

Source: https://www.youtube.com/watch?v=GWWNlxr0Qu8&feature=youtu.be

**Branch(es):**
- [x ] master
- [ ] other

**Hercules rev. hash/commit:** 

Git revision src:

[//]: # (Copy the first 3 lines of the login-server, char-server or map-server startup.)
[//]: # ( [Info]: Hercules 64-bit for Mac OS X )
[//]: # ( [Info]: Git revision src: 'a5918b329ca0826b04dca32ede783586403f58db' )
[//]: # ( [Info]: Git revision scripts: 'a5918b329ca0826b04dca32ede783586403f58db' )

### Operating System

[//]: # (Mac OS X 10.12.3 16D32 [x86_64])
[//]: # (Thank you for adhering to this process! It ensures your issue is resolved quickly and that neither your nor our time is needlessly wasted.)
[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (For fixes please create a Pull Request.)

### LINKED ISSUE #1926: Trap damage - Reflect
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Issue Prelude

[//]: # (Please complete these mandatory steps and check the following boxes by putting an `x` inside the brackets _before_ filing your issue)

-  [x] I have not modified the source prior to reproducing this issue.
-  [x] I am using the latest version of Hercules.
-  [x] I am aware that this report will be closed or deleted if it becomes obvious that I am stating the false.

### Description
In official server (confirmed on Renewal), Reflect Shield skill and `bShortWeaponDamageReturn` can return the damage of all offensive traps like Blastmine, Claymore Trap etc.

Source: https://forums.warpportal.com/index.php?/topic/221853-can-trap-damage-be-reflected/
Official Video - Philippine Ragnarok Online:
https://youtu.be/j6Uk47BLZEY

**Branch(es):**
- [x] master

**Hercules rev. hash/commit:** 

Git revision src:

[//]: # (Copy the first 3 lines of the login-server, char-server or map-server startup.)
[x]: # ( [Info]: Hercules 64-bit for Mac OS X )
[x]: # ( [Info]: Git revision src: 'a5918b329ca0826b04dca32ede783586403f58db' )
[x]: # ( [Info]: Git revision scripts: 'a5918b329ca0826b04dca32ede783586403f58db' )

### Operating System

[//]: # (Mac OS X 10.12.3 16D32 [x86_64])
[//]: # (Thank you for adhering to this process! It ensures your issue is resolved quickly and that neither your nor our time is needlessly wasted.)
[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (For fixes please create a Pull Request.)

### PR DESCRIPTION
**Pull request by guilherme-gm**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)
These commits fixes two issues related to traps.

- The first one allows traps that causes damage to be reflected by skills/cards that gives physical attack reflection (e.g. Reflect Shield and High Orc Card). This feature can be turned off in skill.conf to get old effect back.
As we don't have information about some 3rd classes traps I let them commented out until we're sure they can also be reflected.

- The second issue refers to Blastimine and Claymore Trap damage behavior. The damage caused by these traps must be split by the number of enemies affected by them.

**Affected Branches:** 

[//]: # (Master? Slave?)
master

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)
* Fixes #1926 (First Issue)
* Fixes #1900 (Second Issue)

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

Special thanks to @kyeme

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2182)

File: src/map/battle.c [hunk 1]
```diff
@@ -4339,6 +4339,27 @@ static struct Damage battle_calc_misc_attack(struct block_list *src, struct bloc
 			}
 		break;
 	}
+	
+	if (battle_config.trap_reflect == true) {
```
Comment (4144) [inline src/map/battle.c]: this code with if and switch better move into separate function

File: src/map/skill.c [hunk 2]
```diff
@@ -16864,7 +16865,11 @@ static int skill_trap_splash(struct block_list *bl, va_list ap)
 			}
 			/* Fall through */
 		default:
-			skill->attack(skill->get_type(sg->skill_id),ss,src,bl,sg->skill_id,sg->skill_lv,tick,0);
+			// if trap damage should be split among targets, count them
+			if (skill->get_nk(sg->skill_id)&NK_SPLASHSPLIT)
```
Comment (4144) [inline src/map/skill.c]: need space before ``&`` and after it

File: db/re/skill_db.conf [hunk 3]
```diff
@@ -4520,6 +4520,7 @@ skill_db: (
 	Element: "Ele_Wind"
 	DamageType: {
 		SplashArea: true
+		SplitDamage: true
```
Comment (EyesOfAHawk) [inline db/re/skill_db.conf]: need to add at the top of skill_db.conf what SplitDamage is, boolean etc.

Comment (guilherme-gm) [reply db/re/skill_db.conf]: it's there already :) https://github.com/HerculesWS/Hercules/blob/a3c811a82712fa80c1a8481a54561fb38390638a/db/re/skill_db.conf#L91

**Pull request merged**